src/borg/_hashindex.c

```diff
@@ -562,6 +562,16 @@ hashindex_set(HashIndex *index, const void *key, const void *value)
                 if(!hashindex_resize(index, index->num_buckets)) {
                     return 0;
                 }
+                /* we have just built a fresh hashtable and removed all tombstones,
+                 * so we only have EMPTY or USED buckets, but no DELETED ones any more.
+                 */
+                idx = start_idx = hashindex_index(index, key);
+                while(!BUCKET_IS_EMPTY(index, idx)) {
+                    idx++;
+                    if (idx >= index->num_buckets){
+                        idx -= index->num_buckets;
+                    }
+                }
```

Comment:
This might explain some of the IndexErrors reported over time. The change seems correct to me. It may be debatable whether a recursive call to hashindex_set could be preferable, but imho that is not the case here.

Reply:
This bug was introduced in c5cd882 and affects 1.1, but not 1.0 because there has been no stable 1.0 release since it was committed.

Reply:
Actually this is in 1.0.11, Github just didn't pick that up for some reason.

```
git tag --contains c5cd882
1.0.11
```

Reply:
yeah, see there: https://github.com/borgbackup/borg/issues/4829#issuecomment-593115703